integrationtest/mapper/orm-coordination-database-polling/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/coordination/databasepolling/DatabasePollingAutomaticIndexingBackendFailureIT.java

```diff
@@ -116,8 +104,66 @@ public void backendFailure() {
 		checkId2EntityEventFailure( entityFailure );
 	}
 
+	@Test
+	public void backendFailure_retryAfter() {
+		setup( 3 );
+		withinTransaction( sessionFactory, session -> {
+			IndexedEntity entity1 = new IndexedEntity();
+			entity1.setId( 1 );
+			entity1.setIndexedField( "initialValue" );
+			session.persist( entity1 );
+
+			IndexedEntity entity2 = new IndexedEntity();
+			entity2.setId( 2 );
+			entity2.setIndexedField( "initialValue" );
+			session.persist( entity2 );
+
+			IndexedEntity entity3 = new IndexedEntity();
+			entity3.setId( 3 );
+			entity3.setIndexedField( "initialValue" );
+			session.persist( entity3 );
+
+			CompletableFuture<?> failingFuture = new CompletableFuture<>();
+			failingFuture.completeExceptionally( new SimulatedFailure( "Indexing work #2 failed!" ) );
+
+			backendMock.expectWorks( IndexedEntity.INDEX )
+					.createAndExecuteFollowingWorks()
+					.add( "1", b -> b
+							.field( "indexedField", "initialValue" )
+					)
+					.add( "3", b -> b
+							.field( "indexedField", "initialValue" )
+					)
+					.createAndExecuteFollowingWorks( failingFuture )
+					.add( "2", b -> b
+							.field( "indexedField", "initialValue" )
+					);
+
+		} );
+		backendMock.verifyExpectationsMet();
+
+		long timeOfTheException = System.currentTimeMillis();
+
+		backendMock.expectWorks( IndexedEntity.INDEX )
+				.createAndExecuteFollowingWorks()
+				// retry (succeeds):
+				.createAndExecuteFollowingWorks()
+				.addOrUpdate( "2", b -> b
+						.field( "indexedField", "initialValue" )
+				);
+		backendMock.verifyExpectationsMet();
+
+		long timeOfTheProcess = System.currentTimeMillis();
+
+		// retry only once
+		assertThat( failureHandler.entityFailures.get( 2 ) ).hasSize( 1 );
+		// time is at least 3 sec
+		assertThat( timeOfTheProcess - timeOfTheException ).isGreaterThan( 3000 );
```

Comment:
Technically this *could* be less that 3000 if the first processing happens before `long timeOfTheException =` is executed.

To be sure we're not introducing flaky tests, could you please set `timeOfTheException` at the very end of the transaction that creates the entities? You may need an `AtomicLong` or a `long[]` to store the value, since that code is within a lambda.